packages/block-editor/src/components/global-styles/use-global-styles-output.js

```diff
@@ -1188,6 +1204,26 @@ export function useGlobalStylesOutputWithConfig( mergedConfig = {} ) {
 					isGlobalStyles: true,
 				} );
 			}
+
+			/* CSS for block style variations */
+			if ( mergedConfig.styles.blocks[ blockType.name ]?.variations ) {
+				Object.entries(
+					mergedConfig.styles.blocks[ blockType.name ]?.variations
+				).forEach( ( variation ) => {
+					if ( variation[ 1 ].css ) {
+						const variationSelector =
+							blockSelectors[ blockType.name ]
+								.styleVariationSelectors[ variation[ 0 ] ];
```

Comment:
We should probably add a version of the "not" selector in here, otherwise styles applied to the default variation won't appear in the editor.